webpack.demo.js

```diff
@@ -0,0 +1,54 @@
+var webpack = require('webpack');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  entry: {
+    'angular2': [
+      'zone.js',
+      'reflect-metadata',
```

Comment:
I mean, the point of having webpack is to use webpack. I like to have a simple index.html and let webpack inject all the needed script tags in there.

And about bundled stuff. If we put script tags with development versions of zone.js and reflect-metadata. When I run the uglify, those two dependencies won't be uglified (or I will need to do that in a separate step).

> enables effective caching in browsers

Why is that? I mean, what is the difference if we are not using cache bust?

> avoid unnecessary work on WebPack side

That is not an argument. It is unnecessary work on webpack or unnecessary work on the template side.